packages/office-ui-fabric-react/src/components/Slider/Slider.tsx

```diff
@@ -175,18 +183,32 @@ export class Slider extends BaseComponent<ISliderProps, ISliderState> implements
     const { max, min, step } = this.props;
     const steps: number = (max! - min!) / step!;
     const sliderPositionRect: ClientRect = this.refs.sliderLine.getBoundingClientRect();
-    const sliderLength: number = sliderPositionRect.width;
+    const sliderLength: number = !this.props.vertical ? sliderPositionRect.width : sliderPositionRect.height;
     const stepLength: number = sliderLength / steps;
     let currentSteps: number | undefined;
 
-    if (event.type === 'mousedown' || event.type === 'mousemove') {
-      currentSteps = getRTL() ?
-        (sliderPositionRect.right - (event as MouseEvent).clientX) / stepLength :
-        ((event as MouseEvent).clientX - sliderPositionRect.left) / stepLength;
-    } else if (event.type === 'touchstart' || event.type === 'touchmove') {
-      currentSteps = getRTL() ?
-        (sliderPositionRect.right - (event as TouchEvent).touches[0].clientX) / stepLength :
-        ((event as TouchEvent).touches[0].clientX - sliderPositionRect.left) / stepLength;
+    if (!this.props.vertical) {
+      if (event.type === 'mousedown' || event.type === 'mousemove') {
```

Comment:
Say you had this function:

`
function _getLeft(ev) {
const leftPosition;

 switch (ev.type) {
  'mousedown':
  'mousemove':
     leftPosition = event.clientX;
     break;
  'touchstart':
  'touchmove':
    leftPosition = ev.touches[0].clientX;
    break;
 }
 return leftPosition;
}
`

Now using this, your logic would be significantly less repeat code. That's just an example, and I bet you could extract more here.